### PR TITLE
Add functionality to embedded python code for PYACTION: open/close/shut a well without specifying a report step

### PR DESCRIPTION
**Pull request by lisajulia**
Add functionality to embedded python code for PYACTION: open/close/shut a well without specifying a report step; the report step will then be the current report step

### CODE REVIEW

File: opm/input/eclipse/Schedule/Schedule.cpp [hunk 1]
```diff
@@ -701,16 +701,30 @@ void Schedule::iterateScheduleSection(std::size_t load_start, std::size_t load_e
     void Schedule::shut_well(const std::string& well_name, std::size_t report_step) {
         this->internalWELLSTATUSACTIONXFromPYACTION(well_name, report_step, "SHUT");
     }
+    void Schedule::shut_well(const std::string& well_name) {
+        this->internalWELLSTATUSACTIONXFromPYACTION(well_name, this->current_report_step, "SHUT");
+    }
```
Comment (bska) [inline opm/input/eclipse/Schedule/Schedule.cpp:706]: This has a tiny bit of duplication&ndash;i.e., the string `"SHUT"`.  Would it make sense to implement the function as
```C++
void Schedule::shut_well(const std::string& well_name) {
    this->shut_well(well_name, this->current_report_step);
}
```
instead, to remove that source of **potential** inconsistency?  Similar for the other `*_well()` functions too.

File: opm/input/eclipse/Schedule/Schedule.cpp [hunk 2]
```diff
@@ -701,16 +701,30 @@ void Schedule::iterateScheduleSection(std::size_t load_start, std::size_t load_e
...
+    void Schedule::stop_well(const std::string& well_name) {
+        this->internalWELLSTATUSACTIONXFromPYACTION(well_name, this->current_report_step, "STOP");
+    }
 
     void Schedule::internalWELLSTATUSACTIONXFromPYACTION(const std::string& well_name, std::size_t report_step, const std::string& wellStatus) {
+        if (report_step < this->current_report_step) {
+            throw std::invalid_argument("Well status change for past report step " + std::to_string(report_step) + " requested, current report step is " + std::to_string(this->current_report_step) + ".");
+        } else if (report_step >= this->m_sched_deck.size()) {
+            throw std::invalid_argument("Well status change for report step " + std::to_string(report_step) + " requested, this exceeds the total number of report steps, being " + std::to_string(this->m_sched_deck.size() -1) + ".");
+        }
```
Comment (bska) [inline opm/input/eclipse/Schedule/Schedule.cpp:727]: We already `#include` [`<fmt/format.h>`](https://github.com/OPM/opm-common/blob/e3276a36c76ff7d5e413ac13f87f660c823125f0/opm/input/eclipse/Schedule/Schedule.cpp#L110) in this file, so I think I'd prefer to use that over `std::to_string()` here.

**Review by bska — COMMENTED** [review]
Other than a couple of nits here, I think this looks good.

**Comment by lisajulia** [comment]
jenkins build this please

**Comment by blattms** [comment]
Please indicate somewhere in the title or description that this about PYACTION. Otherwise it will be really hard for others to see what this about.

**Comment by lisajulia** [comment]
Yes, thanks!

> Please indicate somewhere in the title or description that this about PYACTION. Otherwise it will be really hard for others to see what this about.

Yes, thanks!

**Comment by lisajulia** [comment]
@bska Can you have another look and possibly merge? Thanks!

**Review by bska — APPROVED** [review]
Thanks a lot for the updates&ndash;this looks good to me now.  I'll merge into master.

**Pull request merged**